pkg/server/createExperiment.go

```diff
@@ -143,3 +151,19 @@ func ProcessCreateQueue(q chan Request) {
 		go processRequest(&req)
 	}
 }
+
+func SaveRecommendationData(req *Request, experimentName string) (*RecommendationIDMap, *ClusterNamespaceMap) {
+
+	//for  recommendationIDmap:
+	var recommendationIDMap = RecommendationIDMap{RecommendationID: make(map[string]string)}
+	var clusterNamespaceMap = ClusterNamespaceMap{ClusterNamespace: make(map[string]string)}
+
+	reqParts := strings.Split(req.RequestName, "_")
+	recid := reqParts[3]
+	clusterNamespace := fmt.Sprintf("%s_%s", reqParts[1], reqParts[2])
```

Comment:
I've seen this pattern in previous reviews.  Consider using a common function to parse the request name string.

```go
func parseRequestName(reqName string) (id string, cluster string, namespace string) {
	reqParts := strings.Split(reqName, "_")
	id = reqParts[3]
	cluster = reqParts[2]
	namespace = reqParts[1]
	
	// TODO: Include reqParts[0]
	// TODO: Consider adding some validation for these fields.
	// TODO: Consider using a struct for better grouping.
	return id, cluster, namespace
}

Reply:
Found the struct below.  Use the `GetRecommendation` object here.

Reply:
GetRecommendation object is reserved for the user's inputs from getRecommendations call. SaveRecommendationsData is before that call is made.

Reply:
But will note for next PR to add a separate func to parse - Thank you!